tool/tctl/common/resource_command.go

```diff
@@ -1166,6 +1185,25 @@ func (rc *ResourceCommand) getCollection(ctx context.Context, client auth.Client
 			return nil, trace.NotFound("database server %q not found", rc.ref.Name)
 		}
 		return &databaseServerCollection{servers: out}, nil
+	case types.KindKubeServer:
+		servers, err := client.GetKubernetesServers(ctx)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		if rc.ref.Name == "" {
+			return &kubeServerCollection{servers: servers}, nil
+		}
+
+		var out []types.KubeServer
+		for _, server := range servers {
+			if server.GetName() == rc.ref.Name || server.GetHostname() == rc.ref.Name {
```

Comment:
It looks that Hostname is alway empty during Heartbeat call: https://github.com/gravitational/teleport/pull/14389/files#diff-f3d780145cae3e5870d7c2de527ca63e9109f20c9b3fe173f6c989bc86d6d826R248-R254
wonder why we need the `server.GetHostname() == rc.ref.Name` check  ?

Reply:
The hostname shouldn't be always empty since it is defined
https://github.com/gravitational/teleport/blob/0737d8e9bf3a810b4efa6e2083a22622faf3ef70/lib/kube/proxy/server.go#L226-L231

And the listener is not nil at that point

Reply:
AH, yea I missed this part.